### ISSUE TITLE
Make text selection configurable

### ISSUE THREAD

**Issue by barinbritva** [reactjs/react-autocomplete#111]
Autocomplete is fully not usable on mobile devices. Entering a new symbol clears selection, but not adds letter. Need to type letter twice.

![ezgif com-video-to-gif 1](https://cloud.githubusercontent.com/assets/4758362/15648700/efa94878-2674-11e6-9b6c-91ed43982a7b.gif)

On some devices after second typing, letter duplicated.

**Comment by barinbritva** [comment]
Something like this: #112

**Comment by CMTegner** [comment]
I'd like to figure out why this is happening instead of adding a band-aid. Could you provide any more info about the environment you're seeing this problem in? I think the version running on reactcommunity.org is slightly outdated, so it would be great if you could provide an example with the latest version.

**Comment by barinbritva** [comment]
It reproduces on version 1.0.0-rc5. All devices: ios, android, windows phone.

**Comment by CMTegner** [comment]
I wonder if perhaps the native autocomplete is getting in the way (#93). Could you test the latest version from git and see if you can still reproduce?

**Comment by meetmatt** [comment]
@CMTegner I was able to reproduce the issue as well and `autoComplete: 'off'` makes no difference.
+1 for configuration flag to turn it off completely, because in current state it makes it nearly unusable for mobile users.

**Comment by sprjr** [comment]
Since this is making it entirely unusable for mobile users (or a sizeable % of devices for sure) I added this to the next patch release, and I think it'd be great if we could include a proper fix for this in that milestone.

@CMTegner I agree that the fix we do find would be best served to not be a bandaid if possible.

**Comment by ImanMh** [comment]
Is it ready yet?

**Comment by sprjr** [comment]
This should be resolved via #152, we'll put the selection typeahead on a feature enhancement and see when we can get that added back in while working for mobile as well.

Will try to have the patch release out soon (a day or two at latest, if not today)

**Comment by ImanMh** [comment]
I downloaded v0.1.4 but it still highlights the ahead text. Do I have to change any of options?

**Comment by CMTegner** [comment]
The latest official release is [1.1.0 on npm](https://www.npmjs.com/package/react-autocomplete).

**Comment by ImanMh** [comment]
@CMTegner v1.1.0 did not work for me. Do I have to change any of my own code or downloading new version is enough?

**Comment by CMTegner** [comment]
If you're upgrading from 0.1.4 you should consult the [changelog](https://github.com/reactjs/react-autocomplete/blob/master/CHANGELOG.md) to see what's changed. The only breaking change is that you now maintain `props.value` yourself, i.e. the component no longer holds this value in it's state. Compare the [examples](https://github.com/reactjs/react-autocomplete/tree/master/examples) to your own code and see how they differ.

**Comment by ImanMh** [comment]
@CMTegner  I was using ```1.0.1``` and I upgraded to ```1.1.0```. Is there anything else I should do?

**Comment by CMTegner** [comment]
No, there shouldn't be. What seems to be the problem?

**Comment by ImanMh** [comment]
I looked at my ```node_modules/react-autocomplete/package.json``` and version field  is ``` "version": "1.1.0"``` so I'm sure the update has accrued. 
This is what I get when I type something: 

![rac](https://cloud.githubusercontent.com/assets/4482199/17518651/06c083fc-5e5e-11e6-93ed-cdedcb2d25c2.jpg)

Selected area must be gone right?

**Comment by CMTegner** [comment]
Correct, it shouldn't be selecting any text. Can you create a reduced test case?

**Comment by ImanMh** [comment]
I downloaded the library but I had too much trouble with running the examples so I think I copy paste the library from node_modules to my src code directory and resolve this issue manually.

**Comment by Zara603** [comment]
@CMTegner when I update to 1.2.0 the build folder still is 0.2.1 and I'm getting ReferenceError: _ is not defined

**Comment by sprjr** [comment]
@Zara603 - I just noticed an issue with the `build/package.json` being out of _sync_ with the primary package, but the code itself was not changed and was correctly the 1.2.0 code. I verified this by doing `npm i -S react-autocomplete@1.2.0` and inspected the build files to verify they have the `autoHighlight` prop implemented.

I was able to get the component loaded in to a `create-react-app` just to see if I could reproduce the issue. I'm not seeing this `ReferenceError`, is it browser specific?

[ns example](https://ns-vnbgeeupcb.now.sh)

--

_I did publish `1.2.1` just now to help sync the version numbers back up - sorry for the confusion._

@CMTegner fyi about the publish, I also just noticed that the new `.babelrc` changes also impact the `rackt server` which likely explains why @ImanMh wasn't able to run them. I also noticed that the gh-pages branch is 404'ing on the static-data example. Odd, the other two are just fine - and the branch directory looks ok too. I'll try a republish perhaps...
